client/src/components/Common/EditEvent/EditEvent.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from "react";
+import Calender from "../../../Images/calendar_icon.png";
+import Clock from "../../../Images/clock_icon.png";
+import RoomIcon from "../../../Images/room_icon.png";
+import "./editEvent.css";
+import axios from "axios";
+import { Link } from "react-router-dom";
+import Popup from "./Popup";
+class EditEvent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userId: 3,
+      date: "2020-05-23T11:00:00.000Z",
+      roomId: 5,
+      room_name: "Moscow",
+      showPopup: false,
+      event_id: "",
+        event_id: "",
+        event_title: "",
+        event_author: "",
+        event_description: "",
+        event_note: "",
+    };
+  }
+
+  componentDidMount() {
+    let eventId = this.props.match.params.id;
+    axios
+      .get(`/api/edit-event/${eventId}`)
+      .then((res) => {
+        this.setState({            
+          event_id: res.data.event_id,
+          event_title: res.data.event_title,
+          event_author: res.data.event_author,
+          event_description: res.data.event_description,
+          event_note: res.data.event_note
+          
+        });        
+      })
+      .catch((err) => console.log(err));
+  }
+
+  handelChange = (e) => {
+    const name = e.target.name;
+    const value = e.target.value;
+    this.setState({ [name]: value });
+    console.log(this.state);
+    
+  };
+
+  handelSubmit = (e) => {
+    e.preventDefault();
+    const editData = this.state;
+    axios
+      .post('/api/update-event', editData)
+      .then((response) => {
+        if (response.status === 200) {
+          console.log("res :", response);
+          return this.setState({showPopup : !this.state.showPopup})
+
+        }
+        // else  alert('A small error happend in the server in this booking operation!! try again !!')
+      })
+      .catch((err) => console.log(err));
+  };
+
+  render() {
```

Comment:
it seems (though I didn't look properly) that the render method of this component is very similar to the render method of the create event component. If so, it could be extracted into a separate `EventForm` component and reused in here and Create Event, so there is no code duplication..